test/Constraints/diagnostics.swift

```diff
@@ -1183,24 +1183,23 @@ func sr5081() {
   b = a[2...4] // expected-error {{cannot assign value of type 'ArraySlice<String>' to type '[String]'}}
 }
 
+/**
+* TODO(diagnostics):Figure out what to do when expressions are complex and completely broken
```

Comment:
Would it be better to leave this enabled with FIXME'd diagnostics? That way, we'll know when we change the diagnostics, especially if we improve them.

Reply:
Sure, there is just going to be even more diagnostics then before but I can do that.

Reply:
Done!